packages/sanity/src/core/config/prepareConfig.ts

```diff
@@ -495,14 +498,16 @@ function resolveSource({
       config,
     }),
     document: {
-      actions: (partialContext) =>
-        resolveConfigProperty({
+      actions: (partialContext) => {
+        const actions = resolveConfigProperty({
           config,
           context: {...context, ...partialContext},
           initialValue: initialDocumentActions,
           propertyName: 'document.actions',
           reducer: documentActionsReducer,
-        }),
+        })
+        return getStartInCreateSortedActions(actions)
```

Comment:
Ultimately a product question, but this happens only if 1) create integration is enabled in studio config and 2) only for new documents, right? If that's the case it seems fine to me

Reply:
(nvm, I see you called that out already in the PR description)

Reply:
My only thought as someone that doesn't know this part of the code super well is that some folks might wonder why, when trying to set a primary action themselves, they can't do it because the link always goes first.

I will say that it might be that you all have set up an exception for this and the use case, in itself, is an edge case that is out of scope for now.

Reply:
This is entirely intentional, since Product wants to faceplant this feature 🤷 

Devs have two options:
* disable start increate 
* OR filter out the action using actions: prev => {}. 

Either approach will have the same result: No start in create button.
If you know how to do this via the config api I'm all ears though! I would _love_ for this to end up first, but be moveable into the menu.

In a perfect world the plugin api had more expressive power, where we could set priority on actions or something, which would allow overriding when you really want to.

The alternative to this is to do something more drastic like I had to do for "Open in Sanity Create" and "unlink", ie a completly different codepath in structructure. _That_, imo, is even worse of a smell.

Reply:
No no that does sound worse.
I unfortunately don't know how to do it off the top of my head since I don't know that part of the code that well. I'm assuming there is a way of doing: "if there is a custom primary action, then apply that first, otherwise do the open in sanity". 🤔 

However, this is non-blocking because, like you mentioned, folks can disable it if they want or they can filter it out

Reply:
Just doing some final tests, and there is actually a (hacky) way to move the action into the menu.
The gist is: on the root `defineConfig`, intercept the "Start in Sanity Create" action, and remove the action prop! 🙈 

```
  document: {
    actions: (prev) => {
      return prev.map((a) => {
        if (a.action !== ('startInCreate' as any)) {
          return a
        }
        //returs a new function that does not have a action prop
        return (props) => {
          return a(props)
        }
      })
    },
  },
```

![image](https://github.com/user-attachments/assets/14d6a55b-7867-4de3-823d-17077e8d1110)

Yup, its pretty dumb, but it _can_ be done ;)